graphbolt/src/cuda/unique_and_compact_impl.cu

```diff
@@ -32,6 +33,23 @@ struct EqualityFunc {
   }
 };
 
+#define DefineReductionFunction(reduce_fn, name)                              \
+  template <typename scalar_iterator_t>                                       \
+  auto name(const scalar_iterator_t input, int64_t size) {                    \
+    auto allocator = cuda::GetAllocator();                                    \
+    auto stream = cuda::GetCurrentStream();                                   \
+    using scalar_t = std::remove_reference_t<decltype(input[0])>;             \
+    cuda::CopyScalar<scalar_t> result;                                        \
+    size_t workspace_size = 0;                                                \
+    reduce_fn(nullptr, workspace_size, input, result.get(), size, stream);    \
```

Comment:
Add a comment to show this is not get the required size of temporal storage.

Reply:
This is a common pattern in CUB. I will add a CUB_CALL so that we do not have to keep writing this pattern of calls. Opening an issue to track it.

Reply:
#6860